lib/commands/debug/index.js

```diff
@@ -0,0 +1,200 @@
+const path = require('path');
+const fs = require('fs');
+const SmapiClient = require('@src/clients/smapi-client');
+const { AbstractCommand } = require('@src/commands/abstract-command');
+const optionModel = require('@src/commands/option-model');
+const AuthorizationController = require('@src/controllers/authorization-controller');
+const ResourcesConfig = require('@src/model/resources-config');
+const CONSTANTS = require('@src/utils/constants');
+const profileHelper = require('@src/utils/profile-helper');
+const jsonView = require('@src/view/json-view');
+const stringUtils = require('@src/utils/string-utils');
+const Messenger = require('@src/view/messenger');
+const CliError = require('@src/exceptions/cli-error');
+const helper = require('./helper');
+const NodejsDebugFlow = require('./nodejs-debug');
+const PythonDebugFlow = require('./python-debug');
+const JavaDebugFlow = require('./java-debug');
+
+const DEBUG_FLOWS = [
+    JavaDebugFlow,
+    NodejsDebugFlow,
+    PythonDebugFlow
+];
+
+class DebugCommand extends AbstractCommand {
+    name() {
+        return 'debug';
+    }
+
+    description() {
+        return 'debug the skill';
+    }
+
+    requiredOptions() {
+        return [];
+    }
+
+    optionalOptions() {
+        return ['debug-port', 'wait-for-attach', 'skill-code-region', 'debug-region', 'profile', 'debug'];
```

Comment:
i didn't see there are two regions in the description of the cr, and i think it can be confusing to user to understand what each region is for. Should we let user specify one region, and use the logic same as Alexa users - priortize the literally same region, and if absent fallback to default?

Reply:
As per our discussion, we are going to distill the region to s single `--region` param. The skill code region will default to `default` and the debugging connection region to `NA`